### ISSUE TITLE
mnist_distributed.py example doesn't work with TF 1.11

### ISSUE THREAD

**Issue by oliverhu** [linkedin/TonY#42]
mnist_distributed.py example doesn't work with TF 1.11. It complains FLAGS doesn't have `ports` or `logdir`

**Comment by oliverhu** [comment]
@erwa  does this work with 1.12 rc?

**Comment by erwa** [comment]
Will try it out when I get a chance

**Comment by gogasca** [comment]
We will be releasing TF 2.0 early next year. As preparation: We are removing tf.flags and tf.app.
we probably need to define it manually.  Will take a look and submit PR.

```
2018-11-20 08:15:04.552459: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-11-20 08:15:04.559516: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:222] Initialize GrpcChannelCache for job ps -> {0 -> tony-staging-w-0.c.dpe-cloud-mle.internal:41051}
2018-11-20 08:15:04.559583: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:222] Initialize GrpcChannelCache for job worker -> {0 -> localhost:43585, 1 -> tony-staging-w-0.c.dpe-cloud-mle.internal:33949}
2018-11-20 08:15:04.560233: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:381] Started server with target: grpc://localhost:43585
Traceback (most recent call last):
  File "/usr/local/src/jobs/TFJob/src/mnist_distributed.py", line 246, in <module>
    tf.app.run()
  File "/hadoop/yarn/nm-local-dir/usercache/root/appcache/application_1542587994073_0025/container_1542587994073_0025_01_000003/venv/tf112/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/usr/local/src/jobs/TFJob/src/mnist_distributed.py", line 214, in main
    start_tensorboard(FLAGS.working_dir)
  File "/usr/local/src/jobs/TFJob/src/mnist_distributed.py", line 176, in start_tensorboard
    FLAGS.logdir = checkpoint_dir
  File "/hadoop/yarn/nm-local-dir/usercache/root/appcache/application_1542587994073_0025/container_1542587994073_0025_01_000003/venv/tf112/lib/python3.5/site-packages/tensorflow/python/platform/flags.py", line 88, in __setattr__
    return self.__dict__['__wrapped'].__setattr__(name, value)
  File "/hadoop/yarn/nm-local-dir/usercache/root/appcache/application_1542587994073_0025/container_1542587994073_0025_01_000003/venv/tf112/lib/python3.5/site-packages/absl/flags/_flagvalues.py", line 498, in __setattr__
    return self._set_unknown_flag(name, value)
  File "/hadoop/yarn/nm-local-dir/usercache/root/appcache/application_1542587994073_0025/container_1542587994073_0025_01_000003/venv/tf112/lib/python3.5/site-packages/absl/flags/_flagvalues.py", line 374, in _set_unknown_flag
    raise _exceptions.UnrecognizedFlagError(name, value)
absl.flags._exceptions.UnrecognizedFlagError: Unknown command line flag 'logdir'
```

**Comment by oliverhu** [comment]
That would be great, thanks @gogasca

**Comment by gogasca** [comment]
Hi Oliver,
TF 1.12 changed the way of handling flags, 
I opened this to track it: https://github.com/tensorflow/tensorboard/issues/1642

```
from tensorboard.plugins.core import core_plugin
import tensorboard.program as tb_program

def start_tensorboard(logdir):
	tb = tb_program.TensorBoard(plugins=[core_plugin.CorePluginLoader()])
	port = os.getenv('TB_PORT_ENV_VAR', 6006)
	tb.configure(logdir=logdir, port=port)
	tb.launch()
	logging.info("Starting TensorBoard with --logdir=" + logdir)
```

Workaround in official TF pip install 1.12 is to edit tensorboard/program.py and add a "not" in conditional:

```
    for k, v in kwargs.items():
      if not hasattr(flags, k):
        raise ValueError('Unknown TensorBoard flag: %s' % k)
```
We are looking to see if we can release a TB 1.12.1 patch. Otherwise file needs to be edited an then zip the env.

**Comment by oliverhu** [comment]
Hey @gogasca , do you mean the latest released TB is not compatible with latest released TF?

**Comment by gogasca** [comment]
Sorry I mean we will release a patch but this will be in TF itself, currently our TF build process for pip package use same numbering as TB, in this case we will have a TF 1.12.1 with TB 1.12.1.

**Comment by oliverhu** [comment]
Got it. I feel TF evolves really fast (1 patch/rc version a week), we can wait for 1.12.1 if it is coming soon.

**Comment by erwa** [comment]
Thanks for the investigation, @gogasca. I verified that after installing **tf-nightly** and changing `start_tensorboard` to

```
from tensorboard.plugins.core import core_plugin
import tensorboard.program as tb_program

def start_tensorboard(logdir):
	tb = tb_program.TensorBoard(plugins=[core_plugin.CorePluginLoader()])
	port = os.getenv('TB_PORT_ENV_VAR', 6006)
	tb.configure(logdir=logdir, port=port)
	tb.launch()
	logging.info("Starting TensorBoard with --logdir=" + logdir)
```

TensorBoard works with the MNIST distributed code.

**Comment by gogasca** [comment]
TF 1.13 will be released end of the next week

**Comment by gogasca** [comment]
TF 1.13rc0 was released.

**Comment by oliverhu** [comment]
fixed in #216
